rfcs/0045-zkapp-balance-data-in-archive.md

```diff
@@ -0,0 +1,253 @@
+There are several kinds of messages that can be sent using that RPC, but the one
+we're interested in here is the `Breadcrumb_added` message, which contains a block.
+
+Currently, the per-transaction balances (for user commands and internal commands only, not for zkApps)
+are contained in the transaction statuses of transactions contained in a block. The
+transactions are contained in the `Staged_ledger_diff.t` part of the block, which
+is an instance of the type `External_transition.t`.
+
+We'll still want to send blocks using this RPC, but remove some information, and add some
+other information.
+
+The RPC type is unversioned, because the daemon and archive process should come from the
+same build. Hence, the changes proposed here don't require adding versioning.
+
+Information to be removed:
+
+- In `Transaction.Status.t`, the `Applied` constructor is applied to a pair consisting of
+  `Auxiliary_data.t` containing account creation fees, and `Balance_data.t`, containing
+  balances. The fields in those types are all options, because they may not be relevant
+  to a particular transaction. Make the constructor unary, applied to a new type
+  `Account_creation_fees.t` (see below).
+
+- In `Transaction.Status.t`, the `Failed` constructor applied to an instance of `Balance_data.t`.
+  Make the constructor nullary.
+
+- Delete the types `Auxiliary_data.t` and `Balance_data.t`.
+
+- For internal commands, the types `Coinbase_balance_data.t` and `Fee_transfer_balance_data.t`
+  are used in the archive processor to convert the transaction status balance data to a
+  suitable form for those commands. Because per-transaction balances won't be stored, delete
+  those types.
+
+Information to be added:
+
+- The new type `Account_creation_fees.t` is a list of records with
+  fields for public key, amount, and an account role.  The account role can be
+  `Receiver_account`, `Other_party_account`, and so on. This approach
+  gives flexibility for the number of such fees, and avoids the use of
+  options.
+
+- The `Breadcrumb_added` message can contain a list of accounts
+  affected by the block.  Specifically, the list contains a list of
+  `int`, `Account.t` pairs (or perhaps a record with two fields),
+  where the integer is the ledger index, and the account is its
+  ledger state when the block is created. From this list,
+  we can populate the new `accounts` table, and for new accounts, the `public_keys` table.
+  The account information is available in the function `Archive_lib.Diff.Builder.breadcrumb_added`.
+  The staged ledger is contained in the breadcrumb argument, and the block contains transactions,
+  so the accounts affected by the block can be queried from the staged ledger.
+
+Removing the information from transaction statuses changes the
+structure of blocks, which use a versioned type. While the version
+number need not be changed, because the new type will be deployed at a
```

Comment:
The version number has already been incremented for external transitions

Reply:
Right, we don't have to increment again.